### ISSUE TITLE
[NEW DEVICE] IKEA SYMFONISK sound remote gen2 E2123

### ISSUE THREAD

**Issue by Mark-Niemeyer** [xaviml/controllerx#842]
# New Device Support

## Device Information

- Device Model: E2123
- Device Description: SYMFONISK sound remote gen2
- Device Manufacturer: IKEA

#### Actions
from [here](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/6775#issuecomment-1469836141):

`buttonevent` | Button | Action
-- | -- | --
1002 | Play | Press/Release or Hold
2001 | + | Hold (every 0.2s)
2002 | + | Press/Release
3001 | + | Hold (every 0.2s)
3002 | - | Press/Release
4002 | Previous | Press/Release or Hold
5002 | Next | Press/Release or Hold
6000 | . | Initial Press
6001 | . | Hold (only once)
6002 | . | Release (Single Press)
6003 | . | Long Press
6004 | . | Double Press
7000 | .. | Initial Press
7001 | .. | Hold (only once)
7002 | .. | Press/Release (Single Press)
7003 | .. | Long Press
7004 | .. | Double Press

**Comment by yvos** [comment]
Is there anything I can do to deliver data for the Zigbee2mqtt integration.

**Comment by JDFS404** [comment]
Hi @xaviml can this one be supported?

**Comment by xaviml** [comment]
Hi @JDFS404 ,

Yes, it is within the plan to support this device. Do you have one of these controllers?

Regards,
Xavi M.

**Comment by JDFS404** [comment]
Hi @xaviml - I bought one! Let me know if you need some help or anything :)

**Comment by xaviml** [comment]
Hi @JDFS404 ,

I am going through tickets and implementing them as I get some free time. I will try to integrate this device this weekend tops and add it to a beta release.

If you have the programming knowledge, you can also contribute and create PR. The steps to integrate a new device are specified in the CONTRIBUTING.md file.

Thanks,
Xavi M.

**Comment by xaviml** [comment]
As part of this issue, I m also adding support for z2m integration to this controller with the actions specified in the documentation page: https://www.zigbee2mqtt.io/devices/E2123.html

**Comment by xaviml** [comment]
I created following PR if someone wants to review: https://github.com/xaviml/controllerx/pull/966

**Comment by harrismck** [comment]
Not sure this is the right place to ask (didn't think I should make an entirely new issue), but could you potentially add ZHA support as well? If so, I'd be happy to pull your PR branch and test that it works!

**Comment by xaviml** [comment]
Hi @harrismck, It would be better if you create a new PR specifying the actions needed for ZHA.

**Comment by xaviml** [comment]
This feature has been added to ControllerX beta release [v4.27.0](https://github.com/xaviml/controllerx/releases/tag/v4.27.0).

**Comment by JDFS404** [comment]
Hi @xaviml many thanks for integrating this controller! Can I request you to also add it as a `LightController`?

**Comment by xaviml** [comment]
Hi @JDFS404 ,

Yes, not a problem. Would you mind creating a ticket in GitHub about it? I would need to know which actions I should map.

In any case, you can use custom mapping to add your controller through configuration.

https://xaviml.github.io/controllerx/advanced/

Regards, 
Xavi M.

**Comment by JDFS404** [comment]
> Hi @JDFS404 ,
> 
> Yes, not a problem. Would you mind creating a ticket in GitHub about it? I would need to know which actions I should map.
> 
> In any case, you can use custom mapping to add your controller through configuration.
> 
> https://xaviml.github.io/controllerx/advanced/
> 
> Regards, Xavi M.

Looks like the IKEA remote is missing some key functionality, namely the `release` functionality for the `volume_hold` which means for me dimming is out of the question :(. For now I just used `mapping` as a band-aid for non-dimming mappings. Thanks for all the trouble!

**Comment by harrismck** [comment]
While there may be no proper `release` functionality, the controller does send multiple of the `hold_volume_up` event while you're holding, so this solution here enables proper holding!
https://github.com/xaviml/controllerx/pull/1109#issuecomment-2843984174
